src/sentry/sentry_metrics/indexer/cache.py

```diff
@@ -4,11 +4,38 @@
 
 from django.conf import settings
 
+from sentry.sentry_metrics.indexer.models import StringIndexer
 from sentry.utils.cache import cache
 from sentry.utils.hashlib import md5_text
 
 logger = logging.getLogger(__name__)
 
+# todo: add the real names of strings here
+LOCAL_CACHE_KEYS = [
+    "release",
+    "environment",
+    "session",
+]
+
+
+class LocalIndexerCache:
+    def __init__(self) -> None:
+        self.cache: MutableMapping[str, int] = {}
+        self.loaded = False
+
+    def _load_cache(self) -> None:
+        objs = StringIndexer.objects.filter(organization_id=0, string__in=LOCAL_CACHE_KEYS)
+        self.cache = {obj.string: obj.id for obj in objs}
+
+    def get(self, key: str) -> Optional[int]:
+        if not self.loaded:
+            self._load_cache()
+            self.loaded = True
```

Comment:
kept getting a `DB access not allowed` in tests when I tried to load the cache in the init, so using a flag instead for now. I'm assuming this tries to load before the test DB has been created or something like that im not sure